tests/vmi_lifecycle_test.go

```diff
@@ -708,41 +709,43 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
+				node, err := kubevirt.Client().CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred(), "Should patch node successfully")
+
+				Expect(node.Annotations).To(HaveKey(v1.VirtHandlerHeartbeat))
+				lastHeartBeat := node.Annotations[v1.VirtHandlerHeartbeat]
+				timestamp := &metav1.Time{}
+				Expect(json.Unmarshal([]byte(`"`+lastHeartBeat+`"`), &timestamp)).To(Succeed())
+				timeToSet := metav1.NewTime(timestamp.Add(-10 * time.Minute))
+				dataTime, err := json.Marshal(timeToSet)
+				Expect(err).ToNot(HaveOccurred(), "Should marshal to json")
+				dataTime = []byte(strings.Trim(string(dataTime), `"`))
+
+				data := []byte(fmt.Sprintf(`{"metadata": { "labels": { "%s": "true" }, "annotations": {"%s": "%s"}}}`, v1.NodeSchedulable, v1.VirtHandlerHeartbeat, dataTime))
 				_, err = kubevirt.Client().CoreV1().Nodes().Patch(context.Background(), nodeName, types.StrategicMergePatchType, data, metav1.PatchOptions{})
 				Expect(err).ToNot(HaveOccurred(), "Should patch node successfully")
 
-				// Delete vmi pod
-				pods, err := kubevirt.Client().CoreV1().Pods(vmi.Namespace).List(context.Background(), metav1.ListOptions{
-					LabelSelector: v1.CreatedByLabel + "=" + string(vmi.GetUID()),
-				})
-				Expect(err).ToNot(HaveOccurred(), "Should list pods successfully")
-				Expect(pods.Items).To(HaveLen(1), "There should be only one VMI pod")
-				var gracePeriod int64 = 0
-				Expect(kubevirt.Client().CoreV1().Pods(vmi.Namespace).Delete(context.Background(), pods.Items[0].Name, metav1.DeleteOptions{
-					GracePeriodSeconds: &gracePeriod,
-				})).To(Succeed(), "The vmi pod should be deleted successfully")
+				// Note we cannot remove the Pod as the vmi controller also moves the VMI to Failed if Pod disappears
+				// This leads to race condition, killing the process gives us exactly what we want
+				By("killing the virt-launcher")
+				vmiKiller, err := pkillAllLaunchers(kubevirt.Client(), nodeName)
+				Expect(err).ToNot(HaveOccurred(), "Should create vmi-killer pod to kill virt-launcher successfully")
+				watcher.New(vmiKiller).SinceWatchedObjectResourceVersion().Timeout(20*time.Second).WaitFor(context.Background(), watcher.NormalEvent, v1.Started)
 
 				// it will take at least 45 seconds until the vmi is gone, check the schedulable state in the meantime
 				By("marking the node as not schedulable")
-				Eventually(func() string {
+				Eventually(func() map[string]string {
 					node, err := kubevirt.Client().CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
 					Expect(err).ToNot(HaveOccurred(), "Should get node successfully")
-					return node.Labels[v1.NodeSchedulable]
-				}, 20*time.Second, 1*time.Second).Should(Equal("false"), "The node should not be schedulable")
+					return node.Labels
+				}, 10*time.Second, 1*time.Second).Should(HaveKeyWithValue(v1.NodeSchedulable, "false"), "The node should not be schedulable")
```

Comment:
```suggestion
				}).WithPolling(time.Second).WithTimeout(10*time.Second).Should(HaveKeyWithValue(v1.NodeSchedulable, "false"), "The node should not be schedulable")
````

Reply:
Again this is hard for me to sell if I don't do it for the whole file but do I need to do it if I am simply helping CI here?